### PR TITLE
more complete email header collection for w3c archives

### PR DESCRIPTION
**Pull request by npdoty**
These add the additional headers as noted in #380 and tests add a lot more coverage (about 90%), addressing #342.

This includes all of the test changes in PR #344 as well, so we could just merge this one and that PR will be included.

### CODE REVIEW

File: tests/bigbang_test.py [hunk 1]
```diff
@@ -224,3 +224,60 @@ def test_labeled_blockmodel(self):
...
+        shutil.rmtree(TEMP_DIR)
+
+    def test_w3c_archive_message_parsing(self):
+        test_html_path = os.path.join(CONFIG.test_data_path, 'w3crawl-test-message.html')
+        f = open(test_html_path, 'r')
+        html = f.read()
+        f.close()
+        message = w3crawl.W3cMailingListArchivesParser().parsestr(html)
+
+        assert len(message.get_from()) > 0, "message doesn't have a From address"
```
Comment (sbenthall) [inline tests/bigbang_test.py:246]: OH, hmm, I see this is what 'pytest' wants....

**Comment by sbenthall** [comment]
I'm seeing now that unittest has one assertion syntax, but pytest wants another:

https://docs.pytest.org/en/latest/

I'm not sure it matters enough to require either form, but it would be nice to standardize.

What do you think?

**Comment by npdoty** [comment]
I think I would like to standardize for our own purposes, and really wasn't sure as I was writing what I was "supposed" to do.

I think the pytest introspection (where it gives you a useful error message with some context based on the values of the variables) depends on using bare `assert` rather than the self.assertX unittest methods, so I'd be slightly inclined towards that. And that documentation suggests that much of the time we may not want an explanatory message afterwards (as that blocks the introspection).

**Comment by npdoty** [comment]
I agree some consistency would be good, and so I've at least updated this PR so that I'm consistently using bare `assert` syntax (as recommended for pytest) in this module's tests.

**Comment by sbenthall** [comment]
I seem to have left this hanging for months and months.
Should I review this again and merge it?

**Comment by npdoty** [comment]
@sbenthall Yes, this one is updated and ready again to be reviewed. (I closed #344 which has been surpassed by this.)

**Pull request merged**